packages/sync/src/Queue.php

```diff
@@ -1,67 +1,103 @@
 <?php
+/**
+ * The class that describes the Queue for the sync package.
+ *
+ * @package jetpack/sync
+ */
 
 namespace Automattic\Jetpack\Sync;
 
-use Automattic\Jetpack\Sync\Defaults;
-
 /**
  * A persistent queue that can be flushed in increments of N items,
  * and which blocks reads until checked-out buffers are checked in or
  * closed. This uses raw SQL for two reasons: speed, and not triggering
  * tons of added_option callbacks.
  */
 class Queue {
+	/**
+	 * The queue id.
+	 *
+	 * @var string
+	 */
 	public $id;
+	/**
+	 * Keeps track of the rows.
+	 *
+	 * @var int
+	 */
 	private $row_iterator;
 
-	function __construct( $id ) {
-		$this->id           = str_replace( '-', '_', $id ); // necessary to ensure we don't have ID collisions in the SQL
+	/**
+	 * Queue constructor.
+	 *
+	 * @param string $id Name of the queue.
+	 */
+	public function __construct( $id ) {
+		$this->id           = str_replace( '-', '_', $id ); // Necessary to ensure we don't have ID collisions in the SQL.
 		$this->row_iterator = 0;
-		$this->random_int   = mt_rand( 1, 1000000 );
+		$this->random_int   = wp_rand( 1, 1000000 );
 	}
 
-	function add( $item ) {
+	/**
+	 * Add a single item to the queue.
+	 *
+	 * @param object $item Event object to add to queue.
+	 */
+	public function add( $item ) {
 		global $wpdb;
 		$added = false;
-		// this basically tries to add the option until enough time has elapsed that
-		// it has a unique (microtime-based) option key
+		// This basically tries to add the option until enough time has elapsed that
+		// it has a unique (microtime-based) option key.
 		while ( ! $added ) {
 			$rows_added = $wpdb->query(
 				$wpdb->prepare(
 					"INSERT INTO $wpdb->options (option_name, option_value, autoload) VALUES (%s, %s,%s)",
 					$this->get_next_data_row_option_name(),
-					serialize( $item ),
+					serialize( $item ), // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
 					'no'
 				)
 			);
 			$added      = ( 0 !== $rows_added );
 		}
 	}
 
-	// Attempts to insert all the items in a single SQL query. May be subject to query size limits!
-	function add_all( $items ) {
+	/**
+	 * Insert all the items in a single SQL query. May be subject to query size limits!
+	 *
+	 * @param array $items Array of events to add to the queue.
+	 *
+	 * @return bool\WP_Error
+	 */
+	public function add_all( $items ) {
```

Comment:
Do you think it makes sense for some of those methods to be `protected` or `private` instead of `public`?

Reply:
I guess it is useful as some sort of documentation that private ones are not expected to be used.